tool/tctl/common/db_command.go

```diff
@@ -88,24 +95,46 @@ func (c *DBCommand) TryRun(ctx context.Context, cmd string, clientFunc commoncli
 	return true, trace.Wrap(err)
 }
 
-// ListDatabases prints the list of database proxies that have recently sent
-// heartbeats to the cluster.
-func (c *DBCommand) ListDatabases(ctx context.Context, clt *authclient.Client) error {
+func (c *DBCommand) listDatabases(ctx context.Context, clt authclient.ClientI) ([]types.DatabaseServer, error) {
 	labels, err := libclient.ParseLabelSpec(c.labels)
 	if err != nil {
-		return trace.Wrap(err)
+		return nil, trace.Wrap(err)
 	}
 
-	servers, err := client.GetAllResources[types.DatabaseServer](ctx, clt, &proto.ListResourcesRequest{
+	registered, err := client.GetAllResources[types.DatabaseServer](ctx, clt, &proto.ListResourcesRequest{
 		ResourceType:        types.KindDatabaseServer,
 		Labels:              labels,
 		PredicateExpression: c.predicateExpr,
 		SearchKeywords:      libclient.ParseSearchKeywords(c.searchKeywords, ','),
 	})
 	if err != nil {
 		if utils.IsPredicateError(err) {
-			return trace.Wrap(utils.PredicateError{Err: err})
+			return nil, trace.Wrap(utils.PredicateError{Err: err})
 		}
+		return nil, trace.Wrap(err)
+	}
+
+	unregistered, err := c.findUnregisteredDatabases(ctx, clt, registered, labels)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
```

Comment:
this should only be done when unregistered databases are requested since it makes API calls